src/main/java/com/google/sps/servlets/GuidesServlet.java

```diff
@@ -23,16 +23,25 @@
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
         response.setContentType("application/json");
-        response.getWriter().print(gson.toJson(helper.queryAll()));
+        String category = request.getParameter("category");
+
+        if(category == null){
+            response.getWriter().print(gson.toJson(helper.queryAll()));
+            return;
+        }
+        
+        category = Jsoup.clean(category, Whitelist.none());
```

Comment:
What happens if this is an empty string? Does it filter the results, or return all guides?

Reply:
As it is, it returns the guides that have category set to an empty string. But what I am thinking about this is that category is a mandatory field.